roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -401,6 +406,66 @@ static void deviceTwinCallback(DEVICE_TWIN_UPDATE_STATE update_state, const unsi
     json_value_free(root_value);
 }
 
+static bool ReadKeyFromFile(const std::string fileName, std::string &key)
+{
+    //key = NULL;
+    if (fileName.empty())
+    {
+        ROS_ERROR("File name empty.");
+        return false;
+    }
+
+    std::ifstream f(fileName); //taking file as inputstream
+    if(f) {
+        std::ostringstream ss;
+        ss << f.rdbuf(); // reading data
```

Comment:
Can any of this throw? If so, we should catch it and output an appropriate ROS_ERROR